---
title: "RNA Seq - Extra Graphs"
author: "Danny McSweeney"
date: "5/4/2020"
output: html_document
---

```{r}
install.packages("tidyverse")
install.packages("ggplot2")
install.packages("RColorBrewer")
install.packages("plotly")
install.packages("DT")
install.packages("dplyr")
```


```{r}
library(tidyverse)
library(ggplot2)
library(RColorBrewer)
library(plotly)
library(DT)
library(dplyr)
```


```{r}
  if (!requireNamespace('BiocManager', quietly = TRUE))
    install.packages('BiocManager')

  BiocManager::install('EnhancedVolcano')
```


```{r}
  library(EnhancedVolcano)
```

```{r}
CASK <- read.table("CASK_WT1-KO1_alldetected.txt", sep = "\t", header = T)
```

```{r}
Pvalue <- CASK$pvalue
hist(Pvalue,
    main = "P Values Across Transcriptome Under CASK Gene Knockout Conditions",
    col="navy",
    border="white"
    )
```

```{r}
ppi <- 400
png("Pvalue.png", width=13*ppi, height=8*ppi, res=ppi)

Pvalue <- CASK$pvalue
hist(Pvalue,
    main = "P Values Across Transcriptome Under CASK Gene Knockout Conditions",
    col="navy",
    border="white"
    )
```

```{r}
Pvalue <- CASK$pvalue
hist(Pvalue,
  main="P Values of Signficant Differential Gene Expression",
  xlim=c(0, 0.2),
  col="navy",
  border="White"
  )    
```


```{r}
Pvalue <- CASK$pvalue
hist(Pvalue,
  main="P Values of Signficant Differential Gene Expression",
  xlim=c(0, 0.05),
  col="navy",
  border="white",
  breaks=150,
  labels = TRUE
  )    
```

```{r}
ppi <- 400
png("Pvalue_selected.png", width=13*ppi, height=8*ppi, res=ppi)

Pvalue <- CASK$pvalue
hist(Pvalue,
  main="P Values of Significant Differential Gene Expression",
  xlim=c(0, 0.05),
  col="navy",
  border="white",
  breaks=150,
  labels = TRUE
  )    
```

```{r}
CASKfc <- read.table("CASKfc.txt", sep = "\t", header = T)
```


```{r}
FC <- CASKfc$log2FoldChange
hist(FC,
  main="Fold Changes of All Differentially Expressed Genes",
  xlim=c(-11, 11),
  col="#b34a00",
  border="black"
  )    
```

```{r}
ppi <- 400
png("FC_selected.png", width=13*ppi, height=8*ppi, res=ppi)

FC <- CASKfc$log2FoldChange
hist(FC,
  main="Fold Changes of Differentially Expressed Genes With FC>2",
  xlab = "Log2(Fold Change)",
  xlim=c(-11, 11),
  col="#b34a00",
  border="black"
  )    
```



```{r}
InitialMapping1 <- read.table("CASK_InitialMapping_1.txt", sep = "\t", header = T)
```

```{r}
str(InitialMapping1)
```

```{r}
##incomplete attempt
specie <- rep(c("A_WT" , 5) , c("A_KO" , 3) , c("banana" , 3) , c("triticum" , 3) )
condition <- rep(c("Reads_After_Filtering" , "rRNA" , "miRNA" , "tRNA" , "snRNA") , 4)


specie <- c(rep("sorgho" , 3) , rep("poacee" , 3) , rep("banana" , 3) , rep("triticum" , 3) )
condition <- rep(c("normal" , "stress" , "Nitrogen") , 4)
value <- abs(rnorm(12 , 0 , 15))
data <- data.frame(specie,condition,value)

ggplot(InitialMapping1, aes(fill=condition, y=count(), x=specie)) + ) + 
  geom_bar(color="black") + 
  ggtitle("Initial Mapping Analysis of RNA-Seq Read Types") + 
  ylab("Read Count") + 
  xlab("Technical Replicate-Cell Line")
dev.off()
```

```{r}

InitialMapping1 <- read.table("CASK_InitialMapping_1.txt", sep = "\t", header = T)

IM1 <- ggplot() + geom_bar(aes(y = percentage, x = year, fill = product), data = charts.data,
                           stat="identity")

```
  
```{r}

```


```{r}
InitialMapping2 <- read.table("CASK_InitialMapping_2.txt", sep = "\t", header = T)
```
