---
title: "Lab5_McSweeney"
author: "Danny McSweeney"
date: "3/10/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(tidyverse)
library(ggplot2)
library("dplyr")
```


```{r}
 download.file(url="https://ndownloader.figshare.com/files/2292169",
              destfile = "data/portal_data_joined.csv")
```

```{r}
library (tidyverse)
surveys <- read_csv("data/portal_data_joined.csv")
```

```{r}
surveys %>%
    filter(year < 1995) %>%
    select(year, sex, weight)
```

```{r}
surveys_hindfoot_half <- surveys %>%
    filter(!is.na(hindfoot_length)) %>%
    mutate(hindfoot_half = hindfoot_length / 2) %>%
    filter(hindfoot_half < 30) %>%
    select(species_id, hindfoot_half)
```

```{r}
surveys %>%
    count(plot_type) 
```

```{r}
surveys %>%
    filter(!is.na(hindfoot_length)) %>%
    group_by(species_id) %>%
    summarize(
        mean_hindfoot_length = mean(hindfoot_length),
        min_hindfoot_length = min(hindfoot_length),
        max_hindfoot_length = max(hindfoot_length),
        n = n()
    )
```

```{r}
surveys %>%
    filter(!is.na(weight)) %>%
    group_by(year) %>%
    filter(weight == max(weight)) %>%
    select(year, genus, species, weight) %>%
    arrange(year)
```

```{r}
rich_time <- surveys %>%
  group_by(plot_id, year) %>%
  summarize(n_genera = n_distinct(genus)) %>%
  spread(year, n_genera)

head(rich_time)
```

```{r}
rich_time %>%
  gather(year, n_genera, -plot_id)
```

```{r}
surveys_long <- surveys %>%
  gather(measurement, value, hindfoot_length, weight)
```

```{r}
surveys_long %>%
  group_by(year, measurement, plot_type) %>%
  summarize(mean_value = mean(value, na.rm=TRUE)) %>%
  spread(measurement, mean_value)
```




```{r}
##Exercise No. 1
library (tidyverse)
SNPs <- read_tsv("data/23andMe_complete.txt", skip = 14, col_types = 
  cols(chromosome = col_factor()))
select(SNPs, position, genotype)
```

```{r}
##Exercise No. 2
filter(SNPs, chromosome=='MT', genotype != 'A', genotype !='T')
```

```{r}
##Exercise No. 3

SNPs %>%
  group_by(chromosome) %>%
  summarize(max_position = max(position), min_position = min(position),
    n=n())
```

```{r}
##Exercise No. 4
SNPs %>%
  group_by(chromosome) %>%
  summarize(max_position = max(position), min_position = min(position),
    n=n()) %>%
  mutate(density = n/(max_position-min_position))
```

```{r}
##Exercise No. 5
SNPs %>%
  group_by(chromosome) %>%
  summarize(max_position = max(position), min_position = min(position),
    n=n()) %>%
  mutate(density = n/(max_position-min_position)) %>%
  arrange(density)
```

```{r}
##Exercise No. 5
SNPs %>%
  group_by(chromosome) %>%
  summarize(max_position = max(position), min_position = min(position),
    n=n()) %>%
  mutate(density = n/(max_position-min_position)) %>%
  arrange(density)
  print(SNPs, n=25)
  write.csv(SNPs,"data/lab5_McSweeney.csv", row.names = FALSE)
```

